### PR TITLE
Miscellaneous code clean up

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs [hunk 1]
```diff
@@ -9,17 +9,9 @@ namespace Microsoft.Azure.Devices.Client.Transport.AmqpIot
 {
     internal class AmqpIotLinkFactory : ILinkFactory
     {
-        private static readonly AmqpIotLinkFactory s_instance = new AmqpIotLinkFactory();
+        private static readonly Lazy<AmqpIotLinkFactory> s_instance = new(() => new AmqpIotLinkFactory());
```
Comment (drwill-ms) [inline iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs:12]: Rather that rely on not referencing the type before the static ctor gets called, we can use the defined pattern for delayed construction.

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs [hunk 2]
```diff
@@ -659,7 +648,7 @@ public override async Task<long> SendTwinPatchAsync(TwinCollection reportedPrope
                 // responds, this layer can correlate the request.
                 var taskCompletionSource = new TaskCompletionSource<PatchTwinResponse>(TaskCreationOptions.RunContinuationsAsynchronously);
                 _reportedPropertyUpdateResponseCompletions[requestId] = taskCompletionSource;
-                _twinResponseTimeouts[requestId] = DateTime.UtcNow;
+                _twinResponseTimeouts[requestId] = DateTimeOffset.UtcNow;
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs:651]: Consistent use of data type.

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs [hunk 3]
```diff
@@ -760,7 +723,21 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             }
         }
 
-        #endregion Client operations
+        protected private override void Dispose(bool disposing)
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs:726]: public/protected/internal/private method ordering.

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs [hunk 4]
```diff
@@ -1064,7 +1041,54 @@ private async Task HandleIncomingEventMessageAsync(MqttApplicationMessageReceive
...
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private void RemoveOldOperations(object _)
+        {
+            _ = _twinResponseTimeouts
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs:1080]: Linq-ified.

File: provisioning/device/src/Transports/Amqp/AmqpClientConnection.cs [hunk 5]
```diff
@@ -324,25 +327,5 @@ private void CompleteOnException(TransportAsyncCallbackArgs args)
                 _tcs.TrySetException(args.Exception);
             }
         }
-
-        protected private virtual void Dispose(bool disposing)
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/AmqpClientConnection.cs:328]: As this class does not have descendants, we don't need to use this pattern.

File: provisioning/device/src/Transports/Amqp/AmqpClientSession.cs [hunk 6]
```diff
@@ -74,16 +71,9 @@ public AmqpClientLink CreateLink()
             return new AmqpClientLink(this);
         }
 
-        public event EventHandler<LinkCreatedEventArgs> LinkCreated;
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/AmqpClientSession.cs:77]: Unused.

File: provisioning/device/src/Transports/Amqp/AmqpLinkFactory.cs [hunk 7]
```diff
@@ -11,61 +11,24 @@ internal class AmqpLinkFactory : ILinkFactory
...
-            catch (AggregateException ae)
-            {
-                throw ae.GetBaseException();
-            }
+            return settings.IsReceiver()
+                ? new ReceivingAmqpLink(session, settings)
+                : new SendingAmqpLink(session, settings);
         }
 
-        public event EventHandler<LinkCreatedEventArgs> LinkCreated;
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/AmqpLinkFactory.cs:49]: Unused

File: provisioning/device/src/Transports/Amqp/AmqpLinkFactory.cs [hunk 8]
```diff
@@ -11,61 +11,24 @@ internal class AmqpLinkFactory : ILinkFactory
...
         public IAsyncResult BeginOpenLink(AmqpLink link, TimeSpan timeout, AsyncCallback callback, object state)
         {
-            return TaskHelpers.ToAsyncResult(OpenLinkAsync(link, timeout), callback, state);
+            Debug.Fail($"{nameof(AmqpLinkFactory)} open link should not be used.");
+            throw new NotImplementedException();
         }
 
         public AmqpLink CreateLink(AmqpSession session, AmqpLinkSettings settings)
         {
-            AmqpLink link;
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/AmqpLinkFactory.cs:19]: Matching impl in hub device layer. This code path never gets hit. Also for EndLink.

File: provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs [hunk 9]
```diff
@@ -65,6 +64,8 @@ internal ProvisioningTransportHandlerAmqp(ProvisioningClientOptions options)
 
             Argument.AssertNotNull(message, nameof(message));
 
+            using var connectionLostCancellationToken = new CancellationTokenSource();
```
Comment (drwill-ms) [inline provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs:67]: Making the connection close callback local, we can avoid making this a class member.

**Review by abhipsaMisra — APPROVED** [review]

File: provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs (hunk 9 above)
Comment (timtay-microsoft) [reply provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs:67]: I like this, so I'm making this same change in MQTT in #2797 FYI

**Review by timtay-microsoft — APPROVED** [review]

File: provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs (hunk 9 above)
Comment (drwill-ms) [reply provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs:67]: thank you for using it elsewhere :)

**Review by patilsnr — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

**Pull request merged**